### PR TITLE
Ch logout protect middleware

### PR DESCRIPTION
**Pull request by tuyishimejeandamour**
adding auth middleware

and logout feature which redis

### CODE REVIEW

File: src/modules/users/controller/usersController.ts [hunk 1]
```diff
@@ -27,4 +28,16 @@ const getUser = async (req: Request, res: Response) => {
     }
 }
 
-export default { getUsers, getUser }
\ No newline at end of file
+const logout = async (req: Request, res: Response) => {
+    const authHeader = req.headers.authorization;
+    if (!authHeader || !authHeader.startsWith('bearer ')) {
+        return res.status(401).json({ message: 'Invalid authorization header' });
```
Comment (yvettegahamanyi) [inline src/modules/users/controller/usersController.ts]: I think here you were supposed to follow same response format as other functions 
something like this
`responseUtil.handleError(UNAUTHORIZED, "Invalid authorization header");
 return responseUtil.response(res);`

File: src/middlewares/auth.ts
Comment (yvettegahamanyi) [inline src/middlewares/auth.ts:1]: I think responses in this file need the same format as other functions too

Comment (tuyishimejeandamour) [reply src/middlewares/auth.ts:1]: already changed the response formatting

**Comment by key-joshua** [comment]
Change this file name from **`jwt.utils.ts`** to **`jwtUtils.ts`**

**Pull request closed without merging**